scipy/optimize/optimize.py

```diff
@@ -834,12 +834,12 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
         warnflag = 1
         msg = _status_message['maxfev']
         if disp:
-            print('Warning: ' + msg)
+            warnings.warn(msg, UserWarning, 3)
```

Comment:
It would be good to see if `3` for the stacklevel is really needed.

Reply:
Actually here is the relevant part of the Python doc: https://docs.python.org/3/library/warnings.html#warnings.warn
So I guess in this case the default value is ok.

Reply:
@tupui but don't we want it to show up as a warning from the function that the user called rather than private `_minimize_neldermead`?